[Sebastian Tay] iP

Sebtey opened the pull request:
# ~~Duke~~ Veda System
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Veda System allows you to track your ongoing missions and update them, providing a **seamless CLI task management system**. 

It features the following new functionalities:
- Add tasks
- Remove tasks
- Update tasks


With the following types of task now implemented, Veda is now able to differentiate between the types of task: 
1. ToDos: Just a description of the task
2. Events: Tasks that last for a period of time
3. Deadlines: Tasks that are to be completed by a certain point in time

Below is the new suggested implementation for Veda main method

```java
public static void main(String[] args) {
        initialise();

        run();
    }
```

Current implementations:
- [X] Task management logic
- [ ] GUI 💻 


Link to forked repo can be found [here](https://github.com/Sebtey/ip)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Sebtey committed:
Implemented Level 0

Sebtey committed:
echo feature

Sebtey committed:
Add, List feature

Sebtey committed:
Mark as Done

Sebtey committed:
ToDos, Events, Deadlines

Sebtey committed:
Added in simple automated CLI testing

Sebtey committed:
Added in error handling features

Sebtey committed:
Add in deletion of task feature

Sebtey committed:
Implement file systems

Sebtey committed:
Add file reading mechanism

Sebtey committed:
Add autosaving feature

Sebtey committed:
Rename saved tasks file naming convention

Sebtey committed:
Implement date time support for deadline tasks

Sebtey committed:
Add dateTime support for event tasks

Sebtey committed:
OOP TaskList

Sebtey committed:
Code refactoring into OOP

Sebtey committed:
Package the files into repective packages

Sebtey committed:
Add header comments

Sebtey committed:
Merge branch 'add-gradle-support' of https://github.com/Sebtey/ip into add-gradle-support

Sebtey committed:
Set up gradle

Sebtey committed:
Add JUnit testing for Parser and TaskList

Sebtey committed:
Tweek file output name for shadow dependency in build gradle

Sebtey committed:
Add find functionality

Sebtey committed:
Remove dead code

Sebtey committed:
Complete javaDocs for Tasks package

Sebtey committed:
Complete javaDocs for VedaExceptions package

Sebtey committed:
Complete javaDocs for Parser, Storage, TaskList, Ui and Veda

Sebtey committed:
Add checkstyle into build.gradle

Sebtey committed:
Fix coding standard for Parser class

Sebtey committed:
Fix coding standards for Storage class

Sebtey committed:
Fix coding standards for TaskList class

Sebtey committed:
Fix code standards

Sebtey committed:
Merged branch-A-JavaDoc

Sebtey committed:
Merge branch-A-CodingStandard

songgthu commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,77 @@
+    /**
+     * Retrieves input from the user in the command line.
+     *
+     * @return
```
Perhaps you forgot to complete the @return description of javadoc here.

Sebtey replied:
Oh yes, will fix it in my new commits. Thanks! 👍

songgthu commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,174 @@
+     *
+     * @param taskIndex is the index of the task to be unmarked.
+     */
+    public boolean markUndone(int taskIndex) {
```
Maybe you can name this as "markAsUndone" to keep naming consistency with your existing "markAsDone" method?

Sebtey replied:
Yep, will update it as such. Thanks for pointing it out!

songgthu commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,151 @@
+
+                default:
+                    System.out.println("Wrong file format");
+                }
```
Good job for correct formatting for case clauses!

Sebtey replied:
Thanks!

songgthu commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,151 @@
+     * @return true if file is successfully updated.
+     * @throws IOException When writer is called upon to write after being closed.
+     */
+    public boolean updateData(ArrayList<Task> tasks, boolean isAddingTask) throws IOException {
```
Perhaps you can give a more intuitive variable name for isAddingTask? (e.g: isOverwriting, isNewTask)

Sebtey replied:
For this method, the return value is intended to inform the "client" side that the update has been successful. The main purpose is really, just to update the saved data in the disk

songgthu commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,65 @@
+ * @author Sebastian Tay
+ */
+public class Deadline extends Task {
+    //TODO
```
Maybe you can delete this redundant comment here.

songgthu reviewed (commented):
Great job of following the coding standards! 👍

Sebtey committed:
Fix minor code styling issues

Sebtey committed:
Add a functional GUI for Veda

Sebtey committed:
Link Veda logic to GUI

Sebtey committed:
Refactor code for Veda and Parser class

Sebtey committed:
Refactor code for Parser class

Sebtey committed:
Replace magic literals with variables

Sebtey committed:
Refactor Storage class

Sebtey committed:
Refactor tasks package: Replaced magic literals

Tasks package files uses magic literals.

Replacing these with a variable helps improve the readability of the code
and makes it easier to debug.

Let's,
* Add a const within each task class to denote the class symbol
* Use the const in place of the class symbol

Sebtey committed:
Refactor tasks package

Sebtey committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/Sebtey/ip into branch-A-CodeQuality

Sebtey committed:
Add Assertions

Program has rooms for possible bug due to programmer's error such as
assuming that variables are of certain types and values.

Let's instead add some assertions into the files to explicitly declare
the assumptions made.

Sebtey committed:
Merge pull request #1 from Sebtey/branch-A-CodeQuality

Refactor code

The code has numerous rooms for bugs due to the lack of usage of const, magic literals and lack of SLAP.

To fix this, let's do the following changes:

Abstract out lines of codes into functions
Create new exceptions to handle user errors

Sebtey committed:
Merge branch 'master' into branch-A-Assertions

Add in Assertions

We are unable to verify some of our assumptions in the variables being passed.

Let's add in some asserts to ensure that our assumptions hold true.

Sebtey committed:
Merge pull request #2 from Sebtey/branch-A-Assertions

Add in assertions and tidy up code

Sebtey committed:
Add in update functionality for todo tasks

Sebtey committed:
Add edit functionality for deadline and event tasks

Sebtey committed:
Merge pull request #3 from Sebtey/branch-C-Update

Add edit functionality (BCD - Extension)

Sebtey committed:
Fix coding standard issue

Sebtey committed:
Fix GUI

jar file is unable to run and messages gets compressed as chat continues

* messages height set to fixed height and does not vary with chat length
* build.gradle mainClass set to Veda

Sebtey committed:
Fix error messages

Sebtey committed:
Fix message for wrong user input to update function

Sebtey committed:
Fix return statements for error input

Sebtey committed:
Refine GUI

Sebtey committed:
Add basic readme docs

Sebtey committed:
Add updated Readme

Sebtey committed:
Fix formatting of User Guide

Sebtey committed:
Update documentation

Sebtey committed:
Update test cases

Sebtey committed:
Fix output for user input with extra arguments
